### DISCUSSION TITLE
How do I avoid a publicURL from being returned for a file that does not exist?

### DISCUSSION THREAD

**Discussion by cstory1** [supabase/supabase#2859] [Questions] [answered]
I have been facing an issue where I would do a conditional image load for user profile, 

There's a public bucket  'avatars' that saves new file in the order 'folder(uid)/avatar.

Even when a file or folder does not exist, it still returns a publicURL without error.

Is there a better way to approach this or is it a bug?

**Comment by inian** [comment, marked as answer]
Hi @tobiadiks, `getPublicUrl` is just a helper method to construct the public URL of the asset. You will need to make a request to the public URL to see if the asset exists, etc.
